simulation/utils/generate_map.py

```diff
@@ -0,0 +1,241 @@
+import random
+import math
+import matplotlib.pyplot as plt
+from enum import Enum
+import numpy as np
+from scipy.interpolate import splprep, splev
+
+# ************
+class Point:
+    def __init__(self, x=0, y=0):
+        self.x = x
+        self.y = y
+
+    def __str__(self):
+        return '({x},{y})'.format(x=self.x, y=self.y)
+
+    def getPoint(self, x_offset=0):
+        point = (self.x + x_offset, self.y)
+        return point
+
+    def __getitem__(self, key):
+        return self.x if not key else self.y
+
+
+class Map:
+    def __init__(self, starting_point, x_offset, resolution=(1280, 720), seed=None):
+        self.points_before_interpolation = []
+        self.map_points = []
+        self.segments = []
+        self.x_offset = x_offset
+        self.append(starting_point)
+        self.resolution = resolution
+        self.seed = seed
+
+    # ************
+    def append(self, point):
+        self.points_before_interpolation.append(point)
+
+    def __getitem__(self, key):
+        """Get point of the map: e.g. game_map[7] ---> (x7,y7)"""
+        if not len(self.map_points):
+            return self.points_before_interpolation[key]
+        else:
+            return self.map_points[key]
+
+    def get_data_as_points(self):
+        """Get map in format [(x0,y0), (x1,y1), (x2,y2), ...]"""
+        return self.map_points
+
+    def get_data_as_segments(self):
+        """Get map in format [((x1,y1), (x1,y1)), ((x1,y1), (x2,y2)), ((x2,y2), (x3,y3)), ...]"""
+        return self.segments
+
+    def get_seed(self):
+        """Get the seed of the map"""
+        return self.seed
+
+    def X(self):
+        """Get list of x coordinates"""
+        xs = [point.x for point in self.points_before_interpolation] if not len(self.map_points) else [point[0] for point in self.map_points]
```

Comment:
Przykro mi to mówić, ale pep8. Linijka zapewne swoje źródło ma w sztuczkach z przedmiotu Język Python, gdzie robimy takie pociągi, ale tutaj jest to raczej nieczytelne. Chyba lepiej będzie to rozbić na normalne if - else